semeio/jobs/ots/ots.py

```diff
@@ -0,0 +1,597 @@
+        Geertsma model.
+
+        :param vintage_pairs:
+        """
+
+        if len(vintage_pairs) == 0:
+            return 0, []
+
+        vintages = self._vintages_name_date(vintage_pairs)
+        surface = self._surface
+        num_points = len(surface)
+        num_points_calculated = 0
+        du = np.zeros((len(vintages.name), num_points))
+        ts_surfaces = []
+        for iv, (vn, vd) in enumerate(zip(vintages.name, vintages.date)):
+            if self._verbose:
+                print(
+                    "{:%x %X} TS_RPORV: Calculating vintage {:%Y.%m.%d}".format(
+                        dt.now(), vd
+                    )
+                )
+                sys.stdout.flush()
+
+            self.add_survey(vn, vd)
+
+            num_points_calculated = 0
+            for point in range(num_points):
+
+                if not np.isnan(surface.z[point]):
+                    num_points_calculated += 1
+                    r1 = (surface.x[point], surface.y[point], 0)
+                    r2 = (
+                        surface.x[point],
+                        surface.y[point],
+                        surface.z[point] - self._seabed,
+                    )
+                    # subsidence and displacement have opposite sign
+                    # should have minus on dz1 and dz2 here,
+                    # more efficient when calculating ts
+                    dz1 = self.subsidence.eval_geertsma_rporv(
+                        base_survey=vn,
+                        monitor_survey=None,
+                        pos=r1,
+                        youngs_modulus=self._youngs_modulus,
+                        poisson_ratio=self._poisson_ratio,
+                        seabed=self._seabed,
+                    )
+                    dz2 = self.subsidence.eval_geertsma_rporv(
+                        base_survey=vn,
+                        monitor_survey=None,
+                        pos=r2,
+                        youngs_modulus=self._youngs_modulus,
+                        poisson_ratio=self._poisson_ratio,
+                        seabed=self._seabed,
+                    )
+                    du[iv, point] = dz2 - dz1
+
+        for ivp, vp in enumerate(vintage_pairs):
+            if self._verbose:
+                if self._convention == 1:
+                    print(
+                        "{:%x %X} TS_RPORV: Calculating shift"
+                        " {:%Y.%m.%d}-{:%Y.%m.%d} in {} points".format(
+                            dt.now(), vp[0], vp[1], num_points_calculated
+                        )
+                    )
+                if self._convention == -1:
+                    print(
+                        "{:%x %X} TS_RPORV: Calculating shift"
+                        " {:%Y.%m.%d}-{:%Y.%m.%d} in {} points".format(
+                            dt.now(), vp[1], vp[0], num_points_calculated
+                        )
+                    )
+                if num_points_calculated == 0:
```

Comment:
This variable does not change during the loop, I think. Put the logging outside of the loop instead.

Reply:
In fact, it is over-written on each vintage start, and as such will only report on the final vintage pair. Is that really how it should be? This part should probably have been placed on the last line of the previous loop

Reply:
Was bothering my head with this one too. Not really sure, how `num_points_calculated` relates to each vintage pair - as you can see it's reported with each vintage pair. and as such is should be num_points_calcilated for each of the vintage pair (since it's overwritten in current implementation) - So left it like in original, but might be that the answer is like this:
```python
num_points_calculated[vp] 
```

Reply:
So I read a bit on this and it's even more ambiguous than before. Firstly, `num_points_calculated` is computed for each date specified, ie. when having 4 pairs of dates (base + monitor) this will generate 4 surface calculations with 4 distinct  `num_points_calculated`. But after that we do this:
```python
for ivp, vp in enumerate(vintage_pairs):
            if self._convention == 1:
                logging.info(
                    "{:%x %X} TS_SIMPLE: Calculating shift"
                    " {:%Y.%m.%d}-{:%Y.%m.%d} in {} points".format(
                        dt.now(), vp[0], vp[1], num_points_calculated)
)
```
Ie. we report `num_points_calculated` for each pair of surfaces. I need to check what it actually means.

Reply:
I think this is a copy+paste issue. The logging was placed correctly in the original implementation (betting `geertsma_ts` - in which this looks to make sense), and then someone wanted the same logging in a different function and just copy pasted the entire thing, not seeing how it got wrong.

Reply:
It's done via helper function now with `num_points_calculated` brought up before the main loop.